Tests/L0/Maven/_suite.ts

```diff
@@ -1552,4 +1542,111 @@ describe('Maven Suite', function () {
                 done(err);
             });
     })
+
+    /* SonarQube unit tests */
+
+    it('SonarQube common - Build summary is created (dashboard link only when not waiting for server)', () => {
+        // Arrange
+        var mockRunSettings:SonarQubeRunSettings = new SonarQubeRunSettings("projectKey", "serverUrl", "http://dashboardUrl", "taskId", "taskUrl");
+        var mockServer:ISonarQubeServer = new MockSonarQubeServer();
+
+        var analysisMetrics:SonarQubeMetrics = new SonarQubeMetrics(mockServer, mockRunSettings.ceTaskId);
+        var sqReportBuilder:SonarQubeReportBuilder = new SonarQubeReportBuilder(mockRunSettings, analysisMetrics);
+
+        return sqReportBuilder.fetchMetricsAndCreateReport(false)
+            .then((report) => {
+                    assert(report.indexOf('[sqAnalysis_BuildSummary_LinkText >](http://dashboardUrl "projectKey Dashboard")') > -1);
+            });
+    });
+
+    it('SonarQube common - Build summary is created with quality gate failure', () => {
+        // Arrange
+        var mockRunSettings:SonarQubeRunSettings = new SonarQubeRunSettings("projectKey", "serverUrl", "http://dashboardUrl", "asdfghjklqwertyuiopz", "taskUrl");
+        var mockServer:MockSonarQubeServer = new MockSonarQubeServer();
+
+        var analysisMetrics:SonarQubeMetrics = new SonarQubeMetrics(mockServer, mockRunSettings.ceTaskId, 10, 1); // override to a 10-second timeout
+        var sqReportBuilder:SonarQubeReportBuilder = new SonarQubeReportBuilder(mockRunSettings, analysisMetrics);
+
+        // Mock responses from the server for the task and analysis details
+        var taskDetailsJsonObject:any = JSON.parse(fs.readFileSync(path.join(__dirname, 'data/task_details.json'), 'utf-8'));
+        mockServer.setupMockApiCall('/api/ce/task?id=asdfghjklqwertyuiopz', taskDetailsJsonObject);
```

Comment:
The reason I kept the object can be seen in one of the tests where I deliberately return an empty object as a check for tolerance of invalid input.